Documentation/manpages/gsc.rst

```diff
@@ -0,0 +1,349 @@
+:file:`config.yaml.template` as reference.
+
+Command line arguments
+======================
+
+.. option:: --help
+
+   Display usage.
+
+.. program:: gsc-build
+
+:command:`gsc build` -- build GSC
+---------------------------------
+
+Builds a graphenized Docker image of an application image.
+
+Synopsis:
+
+:command:`gsc build` [*OPTION*] <*IMAGE-NAME*> <*APP1.MANIFEST*> [<*APP2.MANIFEST*> ... <*APPN.MANIFEST*>]
+
+.. option:: -d
+
+   Compile Graphene with debug flags and output
+
+.. option:: -L
+
+   Compile Graphene with Linux PAL in addition to Linux-SGX PAL
+
+.. option:: -G
+
+   Build Graphene only and ignore the application image (useful for Graphene
+   development, irrelevant for end users of GSC)
+
+.. option:: IMAGE-NAME
+
+   Name of the application Docker image
+
+.. option:: APP1.MANIFEST
+
+   Application-specific manifest file for the executable entrypoint of the
+   Docker image
+
+.. option:: APPN.MANIFEST
+
+   Application-specific Manifest for the n-th application
+
+
+Application-specific Manifest Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Each application loaded by Graphene requires a separate manifest file.
+:program:`gsc` semi-automatically generates these manifest files. It generates a
+list of trusted files, assumes values for the number of stacks and memory size,
+and generates the chain of trusted children (see below for details). To allow
+specializing each application manifest, :program:`gsc` allows the user to
+augment each generated manifest. In particular this allows to add additional
+trusted or allowed files, and specify a higher memory or number of stacks
+requirement.
+
+:program:`gsc` allows application specific manifest files to be empty or not to
+exist. In this case :program:`gsc` generates a generic manifest file.
+
+Docker Images starting multiple Applications
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Depending on the use case, a Docker container may execute multiple applications.
+The Docker image defines the entrypoint application which could fork additional
+applications. A common pattern in Docker images executes an entrypoint script
+which calls a set of applications. In Graphene the manifest of a parent
+application has to specify all trusted children that might be forked.
+
+We define the parent-child relationship by overestimating the set of possible
+children. Multiple applications are specified as arguments to :program:`gsc`.
+The example below creates a Docker image with three applications. Based on the
+specified chain of applications, :program:`gsc` generates parent-child
+relationships between application ``appi`` and all applications after it in
+the chain (``> appi``). This overestimates the set of trusted children and may
+not map to the actual partent-child relationship. In the example below ``app1``
+may call ``app2`` or ``app3``, and ``app2`` may call ``app3``, but ``app2`` may
+*not* call ``app1``, and ``app3`` may *not* call ``app1`` or ``app2``.
```

Comment:
Does gsc handles below case ?
`sh`(app1) calls `myapp`(app2) and `myapp` calls `exec("/bin/sh", "/bin/sh -c ls")`.
According to above document I assume this case is not supported ? Is this limitation of GSC or graphene SGX ?